### PR TITLE
feat: team environments

### PR DESCRIPTION
**Pull request by nivedin**
Closes hoppscotch/internal-issues#2 #2429 

### Description
This PR introduces a feature that allows users to create team environments. The environment UI is divided into two 'My Environments' and 'Team Environments' same as the collections UI. Users can select the team that they are part of and create, delete, and update the environment variables.

### Checks
- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### Additional Information

TODO
- [x] Update the store environments when the team is selected
- [x] Differentiate the 'team environment' and 'my environment' variable tooltip UI when selected.

### CODE REVIEW

**Comment by 987Nabil** [comment]
Is there anything blocking this feature? Because it would be what empowers us to really use hoppscotch to the fullest 🙂

**Comment by nivedin** [comment]
> Is there anything blocking this feature? Because it would be what empowers us to really use hoppscotch to the fullest 🙂

@987Nabil I'm working on this feature, caught up with some issues will be pushing for PR review by next week 😄

File: packages/hoppscotch-app/helpers/RequestRunner.ts [hunk 1]
```diff
@@ -96,18 +96,29 @@ export const runRESTRequest$ = (): TaskEither<
...
+                      variables: runResult.right.envs.selected,
+                    }
+                  )
+                }
               }
+            } else if (
+              environmentsStore.value.selectedEnvironmentIndex.type ===
+              "TEAM_ENV"
+            ) {
+              console.log("TEAM-VAR-UPDATED")
```
Comment (AndrewBastin) [inline packages/hoppscotch-app/helpers/RequestRunner.ts:121]: No logs :)

File: packages/hoppscotch-app/components/environments/teams/Details.vue [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+      value: "",
+    },
+  })
+}
+
+const removeEnvironmentVariable = (index: number) => {
+  vars.value.splice(index, 1)
+}
+
+const saveEnvironment = () => {
```
Comment (AndrewBastin) [inline packages/hoppscotch-app/components/environments/teams/Details.vue:236]: Save Environment Modal should wait until the actual saving to be done before the modal is closed along with a loading state.
It isn't currently.

**Review by AndrewBastin — CHANGES_REQUESTED** [review]
look into the given corrections

File: packages/hoppscotch-app/helpers/editor/extensions/HoppEnvironment.ts [hunk 3]
```diff
@@ -24,6 +25,10 @@ const HOPP_ENV_HIGHLIGHT_FOUND =
   "bg-accentDark text-accentContrast hover:bg-accent"
 const HOPP_ENV_HIGHLIGHT_NOT_FOUND =
   "bg-red-500 text-accentContrast hover:bg-red-600"
+const HOPP_TEAM_ENV_HIGHLIGHT_FOUND =
+  "bg-accentDark text-primary hover:bg-accent"
+const HOPP_TEAM_ENV_HIGHLIGHT_NOT_FOUND =
+  "bg-red-500 text-primary hover:bg-red-600"
```
[the remark this answers is not in the document]
Comment (nivedin) [reply packages/hoppscotch-app/helpers/editor/extensions/HoppEnvironment.ts]: I have replaced this styling with an my and team env icon on the tooltip.

**Comment by nickwph** [comment]
hello! we have been waiting for this feature, hope this can be merged soon!

File: packages/hoppscotch-app/src/helpers/teams/TeamEnvironmentAdapter.ts [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+
+  private deleteTeamEnvironment(envId: string) {
+    const teamEnvironments = this.teamEnvironmentList$.value.filter(
+      ({ id }) => id !== envId
+    )
+
+    this.teamEnvironmentList$.next(teamEnvironments)
+  }
+
+  private updateTeamEnvironment(updatedEnvironment: TeamEnvironment) {
```
Comment (AndrewBastin) [inline packages/hoppscotch-app/src/helpers/teams/TeamEnvironmentAdapter.ts:139]: Team Environment renames do not trigger updates.

Steps to repro:
1. Use Hoppscotch with 2 users (single user also has issue, but this should work too)
2. Point both of them to the same team and use Team Environments.
3. Create a Team Environment.
4. Then, rename the team environment.
5. Observe that the environment has not been renamed.

I haven't checked if its just the name or other parts too, please do check.

**Review by AndrewBastin — CHANGES_REQUESTED** [review]
`helpers/backend/gql/mutations/DeleteAllVariablesFromTeamEnvironment.graphql` is empty.

Please look into the given issues.

**Review by AndrewBastin — APPROVED** [review]
lgtm :100:

Although we should definitely also think about adding ability to import an environment from personal.

**Pull request merged**